src/handlers/assign.rs

```diff
@@ -739,6 +736,13 @@ async fn find_reviewer_from_names(
     issue: &Issue,
     names: &[String],
 ) -> Result<String, FindReviewerError> {
+    // Fast path for self-assign, which is always allowed.
+    if let [name] = names {
+        if is_self_assign(&name, &issue.user.login) {
+            return Ok(name.clone());
+        }
+    }
```

Comment:
I'm a bit surprised by this.

Intuitively, I would expect only team member (to be more precise, only people who have write access) to be able to be assigned on a pull request. Issue, no problem.

Reply:
That being said, we have the `ReviewerIsPrAuthor` error, so it's not actually possible for *other people* to assign the PR author :laughing: That's inconsistent, yeah.

Reply:
I guess we can do that in a separate PR, asked about it on Zulip.